web/src/engine/websites/CiaoPlus.ts

```diff
@@ -0,0 +1,213 @@
+        const scramble_seed = page.Parameters['seed'] as number;
+        const COL_NUM = 4;
+
+        return DeScramble(blob, async (image, ctx) => {
+
+            function getPieceDimension(width: number, height: number, t: number): TDimension {
+
+                if (width < t || height < t) return null;
+                const s = Cs(t, 8);
+                return width > s && height > s && (width = Math.floor(width / s) * s, height = Math.floor(height / s) * s),
+                {
+                    width: Math.floor(width / t),
+                    height: Math.floor(height / t)
+                };
+            }
+
+            function Cs(e: number, i: number): number {
+                e > i && ([e, i] = [i, e]);
+                const t = (s, o) => s ? t(o % s, s) : o;
+                return e * i / t(e, i);
+            }
+
+            const xs = function* (e: number, i: number) {
+                yield* Is([...Array(e ** 2)].map((s, o) => o), i).map(
+                    (s, o) => ({
+                        source: {
+                            x: s % e,
+                            y: Math.floor(s / e)
+                        },
+                        dest: {
+                            x: o % e,
+                            y: Math.floor(o / e)
+                        }
+                    })
+                );
+            };
+            function Is(e: number[], seed: number) : number[] {
+                const t = Ls(seed);
+                return e.map(o => [t.next().value, o]).sort((o, r) => + (o[0] > r[0]) - + (r[0] > o[0])).map(o => o[1]);
+            }
+
+            const Ls = function* (seed: number) {
+                const i = Uint32Array.of(seed);
+                for (; ;) i[0] ^= i[0] << 13,
+                i[0] ^= i[0] >>> 17,
+                i[0] ^= i[0] << 5,
+                yield i[0];
+            };
+
+            ctx.drawImage(image, 0, 0);
+            const o = getPieceDimension(image.width, image.height, COL_NUM);
+            ctx.clearRect(0, 0, o.width * COL_NUM, o.height * COL_NUM);
+            for (const piece of xs(COL_NUM, scramble_seed ?? 1)) {
+                ctx.drawImage(
+                    image,
+                    piece.source.x * o.width,
+                    piece.source.y * o.height,
+                    o.width,
+                    o.height,
+                    piece.dest.x * o.width,
+                    piece.dest.y * o.height,
+                    o.width,
+                    o.height
+                );
+            }
+
+        });
+    }
+
+    private async CreateRequest(endpoint: URL | string, postData: string[][] = undefined): Promise<Request> {
+        const url = typeof endpoint === 'string' ? new URL(endpoint) : endpoint;
+
+        //CALCULATE THE HASH for the request body OR request parameters
+        const params = postData ? Object.fromEntries(postData) : Object.fromEntries(url.searchParams);
```

Comment:
Consider providing the search params initialiser as object instead of `string[][]` then this same key/value pairs can be used either as get or post parameters depending on the suggested dedicated get/post method wrappers.